pkg/provenance/provenance.go

```diff
@@ -53,460 +68,324 @@ func init() {
-		//HeaderTimeoutPerRequest: time.Second,
+//what happens if I delete the object?
+//need to delete the ObjectFullProvenance for the object
+//add type of ObjectFullProvenance, postgreses for example
+func (o ObjectLineage) SpecHistory() string {
+	s := make([]string, len(o))
+	for v, spec := range o {
+		s[v-1] = spec.String()
 	}
-	//fmt.Printf("%v\n", cfg)
-	c, err := client.New(cfg)
-	if err != nil {
-		log.Fatal(err)
+	return strings.Join(s, "\n")
+}
+func (o ObjectLineage) Bisect(field, value string) string {
+	s := make([]Spec, 0)
+	for _, spec := range o {
+		s = append(s, spec)
 	}
-	kapi := client.NewKeysAPI(c)
-	// set "/foo" key with "bar" value
-	//resourceKey := "/compositions/Deployment/pod42test-deployment"
-	//resourceProv := "{1 ReplicaSet; 2 Pod -1}"
-	resourceKey := string("/compositions/" + resourceKind + "/" + resourceName)
-	fmt.Printf("Setting %s->%s\n",resourceKey, resourceProv)
-	resp, err := kapi.Set(context.Background(), resourceKey, resourceProv, nil)
-	if err != nil {
-		log.Fatal(err)
-	} else {
-		// print common key info
-		log.Printf("Set is done. Metadata is %q\n", resp)
+	sort.Slice(s, func(i, j int) bool {
+		return s[i].Version < s[i].Version
+	})
+	if len(s) == 1 {
+		//check
+		if s[0].AttributeToData[field] == value {
+			return strconv.Itoa(1)
+		}
 	}
-	fmt.Printf("Getting value for %s\n", resourceKey)
-	resp, err = kapi.Get(context.Background(), resourceKey, nil)
-	if err != nil {
-		log.Fatal(err)
-	} else {
-		// print common key info
-		//log.Printf("Get is done. Metadata is %q\n", resp)
-		// print value
-		log.Printf("%q key has %q value\n", resp.Node.Key, resp.Node.Value)
+	for _, v := range s {
+		if v.AttributeToData[field] == value {
+			return strconv.Itoa(v.Version)
+		}
 	}
-	//fmt.Println("Exiting storeProvenance")
+	return strconv.Itoa(-1)
 }
 
-func buildProvenance(parentResourceKind string, parentResourceName string, level int, 
-	compositionTree *[]CompositionTreeNode) {
-	//fmt.Printf("$$$$$ Building Provenance Level %d $$$$$ \n", level)
-	childResourceKindList, present := compositionMap[parentResourceKind]
-	if present {
-		level = level + 1
-		for _, childResourceKind := range childResourceKindList {
-			childKindPlural := KindPluralMap[childResourceKind]
-			childResourceApiVersion := kindVersionMap[childResourceKind]
-			content := getResourceListContent(childResourceApiVersion, childKindPlural)
-			metaDataAndOwnerReferenceList := parseMetaData(content)
-			childrenList := filterChildren(&metaDataAndOwnerReferenceList, parentResourceName)
-			compTreeNode := CompositionTreeNode{
-				Level: level,
-				ChildKind: childResourceKind,
-				Children: childrenList,
-			}
-			*compositionTree = append(*compositionTree, compTreeNode)
+func (o ObjectLineage) SpecHistoryInterval(vNumStart, vNumEnd int) []Spec {
+	s := make([]Spec, vNumEnd-vNumStart+1)
+	for v, spec := range o {
+		if v >= vNumStart && v <= vNumEnd {
+			s[v-vNumStart] = spec
```

Comment:
Why not just s = append(s, spec) ?

Reply:
because in a map it is unordered, I think... so I think it would give me unordered history version 3, version 5, version 1, etc. anyways, I made a sort method so I could probably fix that soon

Reply:
Want me to change it to append(s, spec)?

Reply:
+1